inc/saitypes.h

```diff
@@ -944,6 +944,88 @@ typedef struct _sai_port_eye_values_list_t
     sai_port_lane_eye_values_t *list;
 } sai_port_eye_values_list_t;
 
+/**
+ * @brief System port configuration attributes
+ */
+typedef struct _sai_system_port_config_t
+{
+    uint32_t port_id;
+    uint32_t attached_switch_id;
+    uint32_t attached_core_index;
+    uint32_t attached_core_port_index;
+    uint32_t speed;
+    uint32_t num_voq;
+} sai_system_port_config_t;
+
+/**
+ * @brief System port configuration list
+ */
+typedef struct _sai_system_port_config_list_t
+{
+    /** Number of entries in the list */
+    uint32_t count;
+
+    /** System port configuration list */
+    sai_system_port_config_t *list;
+} sai_system_port_config_list_t;
+
+/**
+ * @brief Fabric port reachability
+ */
+typedef struct _sai_fabric_port_reachability_t
+{
+    /** Remote switch ID (SAI_SWITCH_TYPE_NPU) */
+    uint32_t switch_id;
+
+    /** Remote switch ID is reachable through the fabric port */
+    bool reachable;
+} sai_fabric_port_reachability_t;
+
+/**
+ * @brief Port error status
+ */
+typedef enum _sai_port_err_status_t
+{
+    /** Data Unit CRC Error */
+    SAI_PORT_ERR_STATUS_DATA_UNIT_CRC_ERROR,
+
+    /** Data Unit Size Error */
+    SAI_PORT_ERR_STATUS_DATA_UNIT_SIZE,
+
+    /** Data Unit Misalignment Error */
+    SAI_PORT_ERR_STATUS_DATA_UNIT_MISALIGNMENT_ERROR,
+
+    /** Uncorrectable RS-FEC code word error */
+    SAI_PORT_ERR_STATUS_CODE_GROUP_ERROR,
+
+    /** SerDes Signal is out of sync */
+    SAI_PORT_ERR_STATUS_SIGNAL_LOCAL_ERROR,
+
+    /** Port is not accepting reachability data units */
+    SAI_PORT_ERR_STATUS_NO_RX_REACHABILITY,
+
+    /** Rate of data units with CRC errors passed its threshold */
+    SAI_PORT_ERR_STATUS_CRC_RATE,
+
+    /** Error remote fault indication */
+    SAI_PORT_ERR_STATUS_REMOTE_FAULT_STATUS,
+
+    /** Error status max */
+    SAI_PORT_ERR_STATUS_MAX,
+} sai_port_err_status_t;
```

Comment:
https://github.com/opencomputeproject/SAI/blob/master/inc/saiport.h already have port errors define
i feel this need to be sai_system_port_err_status_t and also enum should have SAI_SYSTEM_*.

Reply:
In this VoQ PR, these errors apply to fabric ports but are generic enough to apply to front panel ports as well.  These port errors don't apply to system ports.